src/xo-app/vm/tab-disks.js

```diff
@@ -50,6 +51,149 @@ import {
+    name: _('vdiNameLabel'),
+    sortCriteria: 'name_label',
+    default: true,
+  },
+  {
+    itemRenderer: vdi => (
+      <Text
+        value={vdi.name_description}
+        onChange={value => editVdi(vdi, { name_description: value })}
+      />
+    ),
+    name: _('vdiNameDescription'),
+    sortCriteria: 'name_description',
+  },
+  {
+    itemRenderer: vdi => (
+      <Size
+        value={vdi.size || null}
+        onChange={size => editVdi(vdi, { size })}
+      />
+    ),
+    name: _('vdiSize'),
+    sortCriteria: 'size',
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const sr = userData.srs[vdi.$SR]
+      return (
+        sr !== undefined && (
+          <XoSelect
+            labelProp='name_label'
+            onChange={sr => migrateVdi(vdi, sr)}
+            predicate={sr => sr.$pool === userData.vm.$pool && isSrWritable(sr)}
+            useLongClick
+            value={sr}
+            xoType='SR'
+          >
+            <Link to={`/srs/${sr.id}`}>{sr.name_label}</Link>
+          </XoSelect>
+        )
+      )
+    },
+    name: _('vdiSr'),
+    sortCriteria: (vdi, userData) => userData.srs[vdi.$SR].name_label,
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vbd = find(
+        userData.vbds,
+        vbd => vbd.VDI === vdi.id && vbd.VM === userData.vm.id
+      )
+      return (
+        <Toggle
+          onChange={bootable => setBootableVbd(vbd, bootable)}
+          value={vbd.bootable}
+        />
+      )
+    },
+    name: _('vbdBootableStatus'),
+    id: 'vbdBootableStatus',
+  },
+  {
+    itemRenderer: (vdi, userData) => {
+      const vm = userData.vm
+      const vbd = find(
+        userData.vbds,
+        vbd => vbd.VDI === vdi.id && vbd.VM === vm.id
+      )
+      return (
+        <StateButton
+          disabledLabel={_('vbdStatusDisconnected')}
+          disabledHandler={connectVbd}
+          disabledTooltip={_('vbdConnect')}
+          enabledLabel={_('vbdStatusConnected')}
+          enabledHandler={disconnectVbd}
+          enabledTooltip={_('vbdDisconnect')}
+          disabled={!(vbd.attached || isVmRunning(vm))}
+          handlerParam={vbd}
+          state={vbd.attached}
+        />
+      )
+    },
+    name: _('vbdStatus'),
+  },
+]
+
+const COLUMNS = filter(COLUMNS_VM_PV, col => col.id !== 'vbdBootableStatus')
+
+const FILTERS = {
+  filterOnlyManaged: 'type:!VDI-unmanaged',
+  filterOnlyRegular: '!type:|(VDI-snapshot VDI-unmanaged)',
+  filterOnlySnapshots: 'type:VDI-snapshot',
+  filterOnlyOrphaned: 'type:!VDI-unmanaged $VBDs:!""',
+  filterOnlyUnmanaged: 'type:VDI-unmanaged',
+}
+
+const GROUPED_ACTIONS = [
+  {
+    disabled: (selectedItems, userData) =>
+      some(selectedItems, vdi => {
+        let disabled = false
+        some(vdi.$VBDs, vbdId => {
+          const vbd = find(userData.vbds, vbd => vbdId === vbd.id)
+          if (vbd.attached) {
+            disabled = true
+            return true
+          }
+        })
+        return disabled
```

Comment:
I think what you're looking for is something like:
```js
some(
  map(flatten(map(selectedItems, '$VBDs')), id => find(userData.vbds, { id }) // All the VBDs objects
  'attached'
)
```
I agree this is not pretty at all so you may need a "vbdId → VBD" map to simplify it. Remember that this will be run pretty often.